services/QuillLMS/client/app/bundles/Staff/styles/style_guide.scss

```diff
@@ -123,3 +123,273 @@
 .short {
   height: 200px;
 }
+
+#color-palette {
+  .color-row {
+    display: flex;
+    flex-wrap: wrap;
+    margin-bottom: 20px;
+  }
+  .element {
+    padding: 10px;
+    max-width: 16%;
+    border-bottom: 1px solid #d6d5d6;
+    border-right: 1px solid #d6d5d6;
+    div {
+      width: 40px;
+      height: 40px;
+    }
+  }
+  .quill-black {
```

Comment:
Have you worked with SCSS maps + `each`? 

This is working scss code, which programmatically generates rules. I used `background-color` as an example:

```css
$quill-colors: quill-green #0AFF0A, quill-red #FF140A;

@each $kvp in $quill-colors {
  .background-color-#{nth($kvp, 1)} {
    background-color: nth($kvp, 2);
  }

}

```

Reply:
Hey @happythenewsad, I have. I felt like it was easier and more readable to define the variables conventionally in the `variables` file. This file just defines classes for use in the display on the `Backpack` page, not sitewide. I think I rewrite this file as follows:

```
$quill-colors: quill-green $quill-green, quill-red $quill-red

@each $kvp in $quill-colors {
  .#{nth($kvp, 1)} {
    background-color: nth($kvp, 2);
  }
}
```

But that `$quill-colors` variable would just get so long that I kind of prefer the way I have it here. If you feel strongly, though, I can change it.

Reply:
Ok - you work with this code more than I do, so let's go with your recommendation.